File: 141. Linked List Cycle/LinkedListCycle.java
 class ListNode {
      int val;
      ListNode next;
      ListNode(int x) {
          val = x;
          next = null;
      }
  }

public class LinkedListCycle {
    public boolean hasCycle(ListNode head) {
        if(head == null)
            return false;

        ListNode former = head.next;
        ListNode latter = null;
        if(former != null){
            latter = former.next;
        }

        while(former != null && latter != null){
            if(former == latter)return true;
            former = former.next;
            latter = latter.next == null ?  null :  latter.next.next;

        }
        return false;
    }
}

File: 141. Linked List Cycle/linked_list_cycle.py
class ListNode:
     def __init__(self, x):
         self.val = x
         self.next = None

class Solution:
    def hasCycle(self, head):
        slow = fast = head
        while fast and fast.next:
            fast = fast.next.next
            slow = slow.next
            if fast is slow:
                return True
        return False




